ckanext/restricted_access/middleware.py

```diff
@@ -31,7 +33,11 @@ def before_request():
         return invalid_request(), 400
 
     if not check_access_by_path():
-        return tk.abort(404, tk._(const.NOT_FOUND_MESSAGE))
+        return tk.abort(
+            ERROR_CODE,
+            ERROR_MESSAGE or tk._(const.NOT_FOUND_MESSAGE)
+            if ERROR_CODE == 404 else ERROR_MESSAGE
+        )
```

Comment:
Overcomplicated logic. The `const.NOT_FOUND_MESSAGE` could be moved into `get_restricted_paths_error_message`. Check any other our extension that has `config.py`.

Reply:
After all, I think there is no need in `const.NOT_FOUND_MESSAGE` at all since it will be set as default CKAN core error message for the appropriate error code if `ckan.restricted.ui_paths.error_message` is empty. Thus I removed this constant and changed the code.